File: Python/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        strList = list(s)
        vowelsSet = set(['A', 'E', 'I', 'O', 'U', 'a', 'e', 'i', 'o', 'u'])
        left = 0
        right = len(s) - 1

        while left < right:
            charLeft = strList[left]
            charRight = strList[right]
            if charLeft in vowelsSet and charRight in vowelsSet:
                strList[left] = charRight
                strList[right] = charLeft
                left += 1
                right -= 1
            elif charLeft in vowelsSet:
                right -= 1
            elif charRight in vowelsSet:
                left += 1
            else:
                left += 1
                right -= 1

        return ''.join(strList)

File: Swift/reverse-vowels-of-a-string.swift
class Solution {
    func reverseVowels(_ s: String) -> String {
        let vowelsSet = Set<Character>(["A", "E", "I", "O", "U", "a", "e", "i", "o", "u"])
        var chars = [Character](s)
        var left = 0
        var right = chars.count - 1

        while left < right {
            let charLeft = chars[left]
            let charRight = chars[right]
            if vowelsSet.contains(charLeft) && vowelsSet.contains(charRight) {
                chars[left] = charRight
                chars[right] = charLeft
                left += 1
                right -= 1
            } else if vowelsSet.contains(charLeft) {
                right -= 1
            } else if vowelsSet.contains(charRight) {
                left += 1
            } else {
                left += 1
                right -= 1
            }
        }

        return String(chars)
    }
}
